File: Greedy/Make it Divisible by 25/Solution.cpp
// Problem Link : https://codeforces.com/problemset/problem/1593/B
#include <bits/stdc++.h>
using namespace std;
int give(const string &n, const string &hehe) {
    int index = hehe.size() - 1;
    int ans = 0;
    for (int i = n.size() - 1; i >= 0; i--) {
        if (n[i] == hehe[index]) {
            index--;
            if (index < 0)
                break;
        } else {
            ans++;
        }
    }
    if (index >= 0) 
        return INT_MAX;
    return ans;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    while (t--) {
        long long x;
        cin >> x;
        string n = to_string(x);
        string hehe[4] = {"00", "25", "50", "75"};

        int ans = INT_MAX;
        for (int i = 0; i < 4; i++) {
            ans = min(ans, give(n, hehe[i]));
        }
        cout << ans << "\n";
    }

    return 0;
}

File: Greedy/Make it Divisible by 25/Solution.java
// Problem Link : https://codeforces.com/problemset/problem/1593/B
import java.util.*;

public class Main {

    public static int give(String n, String hehe) {
        int index = hehe.length() - 1;
        int ans = 0;
        for (int i = n.length() - 1; i >= 0; i--) {
            if (hehe.charAt(index) == n.charAt(i)) {
                index--;
                if (index < 0)
                    break;
            } else
                ans++;
        }

        if (index >= 0)
            return Integer.MAX_VALUE;

        return ans;

    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int t = sc.nextInt();
        while (t-- > 0) {
            int ans = Integer.MAX_VALUE;
            long x = sc.nextLong();
            String n = x + "";
            String[] hehe = { "00", "25", "50", "75" };
            for (int i = 0; i < 4; i++) {
                ans = Math.min(ans, give(n, hehe[i]));
            }
            System.out.println(ans);
        }
        sc.close();
    }
}
